tools/module_dependencies.py

```diff
@@ -120,21 +111,16 @@ def dependency_toml():
 
 def modules_dependent_on(modules: set[str] | list[str]) -> list[str]:
     """Return the set of modules that depend on any of the given modules."""
-    changed_modules = modules
     if not isinstance(modules, (set, list)):
         raise ValueError("`modules` must be set or list")
 
-    A = dependency_graph().astype(bool)
-
-    pkg_mods, pkg_mods_idx = _pkg_modules()
-    pkg_mods_arr: np.typing.NDArray = np.array(pkg_mods, dtype=object)
-
-    all_dependent_mods = []
-    for changed_mod in changed_modules:
-        dependent_mods = pkg_mods_arr[A[:, pkg_mods_idx[changed_mod]]]
-        all_dependent_mods.extend(dependent_mods.tolist())
+    pkg_mods = _pkg_modules()
+    A = dependency_graph()
+    j = np.zeros_like(A[:, 0])  # boolean indices of dependent modules
+    for module in modules:
+        j |= A[:, pkg_mods.index(module)]
```

Comment:
This is what I had in mind with https://github.com/scikit-image/scikit-image/pull/7749#discussion_r1992109159, but feel free to revert.

Reply:
Fine, if it gives the right result; I just don't understand it yet :)

Reply:
`A[:, module]` contains the boolean indices that extracts (from `pkg_mods`) the names of modules that `module` depends on.

We want the names of modules that ANY `module` in `modules` depends on, so we take the OR of all these boolean index columns.

`j` is the resulting boolean index, and we that to extract the names from `pkg_mods`.

Reply:
it's late, clearly